src/ModelInit.jl

```diff
@@ -260,48 +335,71 @@ end
 
 
 @doc """
-Break the images into tiles and initialize the model parameters.
-
-Args:
-  - blob: A Blob containing the images.
-  - mp: Model parameters.
-
-Returns:
-  Updates in place mp's patches and tile sources.
-  Returns a tiled blob.
+Turn a blob and vector of catalog entries into a tiled_blob and model
+parameters that can be used with Celeste.
 """ ->
-function initialize_tiles_and_patches!(
-    blob::Blob, mp::ModelParams; patch_radius=Inf, fit_psf=true)
-  tiled_blob = SkyImages.break_blob_into_tiles(blob, mp.tile_width)
-  initialize_tiles_and_patches!(
-    tiled_blob, blob, mp, patch_radius=patch_radius, fit_psf=fit_psf)
-  tiled_blob
+function initialize_celeste(
+    blob::Blob, cat::Vector{CatalogEntry};
+    tile_width::Int64=typemax(Int64), fit_psf::Bool=true,
+    patch_radius::Float64=-1., radius_from_cat::Bool=true)
+
+  tiled_blob = SkyImages.break_blob_into_tiles(blob, tile_width)
+  mp = initialize_model_params(tiled_blob, blob, cat,
+                               fit_psf=fit_psf, patch_radius=patch_radius,
+                               radius_from_cat=radius_from_cat)
+  tiled_blob, mp
 end
 
 
 @doc """
-Initialize the tiles and patches if you've already tiled your blob
-(e.g. when you are cropping to a single object location)
+Initilize the model params to the given catalog and tiled image.
+
+Args:
+  - tiled_blob: A TiledBlob
+  - blob: The original Blob
+  - cat: A vector of catalog entries
+  - fit_psf: Whether to give each patch its own local PSF
+  - patch_radius: If set less than Inf or if radius_from_cat=false,
+                  the radius in world coordinates of each patch.
+  - radius_from_cat: If true, choose the patch radius from the catalog.
+                     If false, use patch_radius for each patch.
 """ ->
-function initialize_tiles_and_patches!(
-    tiled_blob::TiledBlob, blob::Blob, mp::ModelParams;
-    patch_radius=Inf, fit_psf=true)
-  # Set the model parameters
+function initialize_model_params(
+    tiled_blob::TiledBlob, blob::Blob, cat::Vector{CatalogEntry};
+    fit_psf::Bool=true, patch_radius::Float64=-1., radius_from_cat::Bool=true)
 
   @assert length(tiled_blob) == length(blob)
+  @assert(length(cat) > 0,
+          "Cannot initilize model parameters with no catalog entries")
+
+  # If patch_radius is set by the caller, don't use the radius from the catalog.
+  if patch_radius != -1.
+    radius_from_cat = false
+  end
+  if !radius_from_cat
+    @assert(patch_radius > 0.,
+            "If !radius_from_cat, you must specify a positive patch_radius.")
+  end
+  vp = Array{Float64, 1}[init_source(ce) for ce in cat]
+  mp = ModelParams(vp, sample_prior())
+
   mp.patches = Array(SkyPatch, mp.S, length(blob))
   mp.tile_sources = Array(Array{Array{Int64}}, length(blob))
 
   for b = 1:length(blob)
     for s=1:mp.S
-      mp.patches[s, b] =
-        SkyPatch(mp.vp[s][ids.u], patch_radius, blob[b], fit_psf=fit_psf)
+      if radius_from_cat
```

Comment:
It just needs a coma. I'll fix it.

On Thu, Sep 24, 2015 at 3:32 PM Ryan <notifications@github.com> wrote:

> In src/ModelInit.jl
> <https://github.com/jeff-regier/Celeste.jl/pull/62#discussion_r40381607>:
>
> >    mp.patches = Array(SkyPatch, mp.S, length(blob))
> >    mp.tile_sources = Array(Array{Array{Int64}}, length(blob))
> >
> >    for b = 1:length(blob)
> >      for s=1:mp.S
> > -      mp.patches[s, b] =
> > -        SkyPatch(mp.vp[s][ids.u], patch_radius, blob[b], fit_psf=fit_psf)
> > +      if radius_from_cat
>
> This fails with the error
>
> ERROR: start has no method matching start(::CatalogEntry)
>
> I'm not sure why. Rather than mess with it just at this moment I'm just
> going to change it back.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jeff-regier/Celeste.jl/pull/62/files#r40381607>.
>